js/modules/networkgraph/node-animation.js

```diff
@@ -0,0 +1,150 @@
+                    lineHeightCorrection = ren
+                        .fontMetrics(null, label.element).h;
+                }
+                return {
+                    x: pos.x + (parent.translateX || 0) + padding,
+                    y: pos.y + (parent.translateY || 0) + padding -
+                        lineHeightCorrection,
+                    width: label.width - 2 * padding,
+                    height: label.height - 2 * padding
+                };
+            }
+        };
+        for (i = 0; i < len; i++) {
+            label = labels[i];
+            if (label) {
+                // Mark with initial opacity
+                label.oldOpacity = label.opacity;
+                label.newOpacity = 1;
+                label.absoluteBox = getAbsoluteBox(label);
+            }
+        }
+        // Prevent a situation in a gradually rising slope, that each label will
+        // hide the previous one because the previous one always has lower rank.
+        labels.sort(function (a, b) {
+            return (b.labelrank || 0) - (a.labelrank || 0);
+        });
+        // Detect overlapping labels
+        for (i = 0; i < len; i++) {
+            label1 = labels[i];
+            box1 = label1 && label1.absoluteBox;
+            for (j = i + 1; j < len; ++j) {
+                label2 = labels[j];
+                box2 = label2 && label2.absoluteBox;
+                if (box1 &&
+                    box2 &&
+                    label1 !== label2 && // #6465, polar chart with connectEnds
+                    label1.newOpacity !== 0 &&
+                    label2.newOpacity !== 0) {
+                    if (isIntersectRect(box1, box2)) {
+                        (label1.labelrank < label2.labelrank ? label1 : label2)
+                            .newOpacity = 0;
+                    }
+                }
+            }
+        }
+        var isLabelAffected = false;
+        // Hide or show
+        labels.forEach(function (label) {
+            var complete, newOpacity;
+            if (label) {
+                newOpacity = label.newOpacity;
+                if (label.oldOpacity !== newOpacity) {
+                    // Make sure the label is completely hidden
+                    // to avoid catching clicks (#4362)
+                    if (label.alignAttr && label.placed) { // data labels
+                        if (newOpacity) {
+                            label.removeClass('highcharts-simulation-dataLabels');
+                            label.addClass('highcharts-simulation-dataLabels-fadeout');
+                            if (!styledMode) { // <--- !styledMode :
+                                label.css({
+                                    opacity: 1,
+                                    transition: 'opacity 2000ms'
+                                });
+                            }
+                        }
+                        else {
+                            complete = function () {
```

Comment:
These lines are similar to IF, except place it in the complete. I would consider to extract it as an internal function and call in IF or in Complete.

function should be also with param (opacity 0 or 1 and adapt classes).

Reply:
Thanks! I have refactored this part :)